src/inject/dynamic-theme/index.ts

```diff
@@ -370,6 +404,7 @@ export function removeDynamicTheme() {
     shadowRootsWithOverrides.clear();
     forEach(styleManagers.keys(), (el) => removeManager(el));
     forEach(document.querySelectorAll('.darkreader'), removeNode);
+    removeAdoptedStyleSheets();
```

Comment:
Do `adoptedStyleManagers.destroy(); adoptedStyleManagers.splice(0);` here. Also clean adopted sheets cache here (a function that clears weak maps).